snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -470,9 +470,36 @@ def prime(self, force=False) -> None:
                 _migrate_files(system, system_dependency_paths, '/',
                                self.primedir, follow_symlinks=True)
 
+        # TODO: base snap support
+        core_path = common.get_core_path()
+
+        def mangle_library_path(library_path: str, elf_file_path: str) -> str:
+            # If the path is is in the core snap, use the absolute path,
+            # if the path is primed, use $ORIGIN, and last if the dependency
+            # is not anywhere return an empty string.
+            #
+            # Once we move away from the system library grabbing logic
+            # we can move to a smarter library capturing mechanism.
+            library_path = library_path.replace(self.installdir, self.primedir)
+            if library_path.startswith(core_path):
+                return os.path.dirname(library_path)
+            elif (library_path.startswith(self.primedir) and
+                  os.path.exists(library_path)):
+                rel_library_path = os.path.relpath(library_path, elf_file_path)
+                rel_library_path_dir = os.path.dirname(rel_library_path)
+                # return the dirname, with the first .. replace with $ORIGIN
+                return rel_library_path_dir.replace('..', '$ORIGIN', 1)
```

Comment:
Why is there no test that checks for this $ORIGIN condition?

Reply:
Because I missed it I guess?